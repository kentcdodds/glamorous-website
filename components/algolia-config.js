import preval from 'preval.macro'
import React from 'react'

export function AlgoliaLink() {
  return (
    <link
      rel="stylesheet"
      href="https://cdn.jsdelivr.net/docsearch.js/2/docsearch.min.css"
    />
  )
}

export function AlgoliaScript() {
  return (
    <script src="https://cdn.jsdelivr.net/docsearch.js/2/docsearch.min.js" />
  )
}

// eslint-disable-next-line no-var vars-on-top
const algoliaFacetFilters = preval`
  const lang = require('./utils/locale')
  const {fallbackLocale} = require('../config.json')
  if (lang === fallbackLocale) {
    module.exports = []
  } else {
    module.exports = [\`language:\${lang}\`]
  }
`

function algoliaSettings() {
  if (typeof docsearch === 'undefined') {
    return
  }
  // eslint-disable-next-line no-undef
  docsearch({
    apiKey: 'b5cb8dd730a01fb05e75f21396760fb8',
    indexName: 'glamorous',
    inputSelector: '.algolia_searchbox',
    algoliaOptions: {
      facetFilters: algoliaFacetFilters,
    },
    debug: false,
  })
}

export {algoliaSettings}
