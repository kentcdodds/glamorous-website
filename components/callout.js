import React from 'react'
import glamorous from 'glamorous'
import mdToHTML from './utils/md-to-html'

const Container = glamorous.div(({type}, theme) => ({
  borderLeft: '4px solid',
  borderColor: theme.colors[type],
  backgroundColor: theme.colors[`${type}Light`],
  padding: '12px 22px 8px 28px',
  margin: '18px 0px 26px 0px',
}))

const Title = glamorous.div(({type}, theme) => ({
  color: theme.colors[type],
  fontWeight: 'bold',
}))

function Callout({type = 'info', title, children}) {
  return (
    <Container type={type}>
      {title ?
        <Title type={type}>
          <div dangerouslySetInnerHTML={{__html: mdToHTML(title)}} />
        </Title> :
        null}
      <div>
        <div dangerouslySetInnerHTML={{__html: mdToHTML(children)}} />
      </div>
    </Container>
  )
}

export default Callout
