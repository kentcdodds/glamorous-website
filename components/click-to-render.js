import React, {Component} from 'react'

class ClickToRender extends Component {
  state = {shouldRender: false}
  setRender = () => {
    if (this.state.shouldRender) {
      // if we've already rendered
      // then we'll never reset the render state
      // The original use case was for codesandbox iframes to not have
      // to reload every time you click to expand them.
      return
    }
    setTimeout(() => {
      this.setState({shouldRender: this._details.open})
    })
  }
  render() {
    const {shouldRender} = this.state
    const {render, component, props} = this.props
    return (
      <details ref={n => (this._details = n)} onClick={this.setRender}>
        <summary>{this.props.summary}</summary>
        {shouldRender && render ? render(props) : null}
        {shouldRender && component ?
          React.createElement(component, props) :
          null}
      </details>
    )
  }
}

export default ClickToRender

// I think jsx-a11y is mistaken with <details></details> elements
/*
eslint
  jsx-a11y/no-noninteractive-element-interactions:0,
  jsx-a11y/click-events-have-key-events:0
*/
