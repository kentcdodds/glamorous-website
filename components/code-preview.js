import React from 'react'
import {LiveProvider, LiveEditor, LiveError, LivePreview} from 'react-live'
import glamorous from 'glamorous'

const StyledLiveProvider = glamorous(LiveProvider)({
  margin: '0 auto',
  display: 'flex',
  flexDirection: 'column',
  flexWrap: 'wrap',
  justifyContent: 'space-around',
  maxWidth: '50rem',
})

const StyledLiveEditor = glamorous(LiveEditor)({})

const StyledLivePreview = glamorous(LivePreview)({
  padding: '1rem',
})

const StyledLiveError = glamorous(LiveError)((props, {colors, fonts}) => ({
  color: colors.code,
  fontFamily: fonts.monospace,
  backgroundColor: colors.white,
  flexBasis: '100%',
  width: '100%',
  maxWidth: '100%',
  padding: '1rem',
}))

const CodePreview = props => {
  return (
    <StyledLiveProvider noInline code={props.code} scope={{glamorous}}>
      <StyledLivePreview />
      <StyledLiveError />
      <StyledLiveEditor />
    </StyledLiveProvider>
  )
}

export default CodePreview
