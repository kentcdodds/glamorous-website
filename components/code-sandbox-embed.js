import React from 'react'

export default CodeSandboxEmbed

function CodeSandboxEmbed({title, id}) {
  return (
    <iframe
      title={title}
      src={`https://codesandbox.io/embed/${id}`}
      style={{
        width: '100%',
        height: 500,
        border: 0,
        borderRadius: 4,
        overflow: 'hidden',
      }}
      sandbox="allow-modals allow-forms allow-popups allow-scripts allow-same-origin"
    />
  )
}
