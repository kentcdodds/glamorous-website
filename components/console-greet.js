import React from 'react'
import InlineScript from './inline-script'

export default ConsoleGreet

function ConsoleGreet() {
  return <InlineScript fn={greet} />
}

function greet() {
  const callToUsers =
    '%c\n😎 If you are using glamorous already please add it here 😎'
  const callForContributors = '%c\n👉 We want you to contribute 👈'

  const CONTRIBUTING =
    'https://github.com/paypal/glamorous/blob/master/CONTRIBUTING.md'
  const USERS = 'https://github.com/paypal/glamorous/blob/master/other/USERS.md'

  const GLAMOROUS =
    ' ██████╗ ██╗      █████╗ ███╗   ███╗ ██████╗ ██████╗  ██████╗ ██╗   ██╗███████╗\n' +
    '██╔════╝ ██║     ██╔══██╗████╗ ████║██╔═══██╗██╔══██╗██╔═══██╗██║   ██║██╔════╝\n' +
    '██║  ███╗██║     ███████║██╔████╔██║██║   ██║██████╔╝██║   ██║██║   ██║███████╗\n' +
    '██║   ██║██║     ██╔══██║██║╚██╔╝██║██║   ██║██╔══██╗██║   ██║██║   ██║╚════██║\n' +
    '╚██████╔╝███████╗██║  ██║██║ ╚═╝ ██║╚██████╔╝██║  ██║╚██████╔╝╚██████╔╝███████║\n' +
    ' ╚═════╝ ╚══════╝╚═╝  ╚═╝╚═╝     ╚═╝ ╚═════╝ ╚═╝  ╚═╝ ╚═════╝  ╚═════╝ ╚══════╝\n'

  const gradientFont = `
    font-size: 20px; font-family: "Helvetica",sans-serif;
    background: -webkit-linear-gradient(top, #FFB6C1, #ED5C70);
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
  `

  /* eslint-disable */
  console.log('%c💄 Welcome to 💄', 'font-size: 24px; color: #444;')

  console.log(`%c${GLAMOROUS}`, 'color: #ED5C70; font-family: monospace')

  console.log(callForContributors, gradientFont)
  console.log(`%c${CONTRIBUTING}`, 'font-size: 14px')

  console.log(callToUsers, gradientFont)
  console.log(`%c${USERS}`, 'font-size: 14px')
  /* eslint-enable */
}
