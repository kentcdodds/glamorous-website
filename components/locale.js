import React, {Component} from 'react'
import PropTypes from 'prop-types'

const CONTEXT_NAME = '__locale__'
const supportedLocales = ['en', 'es']
const fallbackLocale = 'en'

class LocaleProvider extends Component {
  static childContextTypes: {[CONTEXT_NAME]: PropTypes.string}
  static getChildContext() {
    return {[CONTEXT_NAME]: this.props.locale}
  }
  render() {
    return this.props.children
  }
}

function withLocale(Comp) {
  class ThemedComponent extends Component {
    state = {locale: 'en'}
    componentWillMount() {
      this.setState({locale: this.context[CONTEXT_NAME]})
    }
    render() {
      return <Comp {...this.props} {...this.state} />
    }
  }

  ThemedComponent.contextTypes = {
    [CONTEXT_NAME]: PropTypes.string,
  }

  return ThemedComponent
}

function getInitialLocaleProps({req: {headers: {host = 'en'} = {}} = {}}) {
  console.log(host)
  const [locale] = host.split('.')
  return Promise.resolve({
    locale: supportedLocales.includes(locale) ? locale : 'en',
  })
}

function getTranslations(locale, path) {
  try {
    return require(`../translations/${locale}/${path}`)
  } catch (error) {
    if (locale === fallbackLocale) {
      throw error
    }
    return getTranslations(fallbackLocale, path)
  }
}

export {LocaleProvider, withLocale, getInitialLocaleProps, getTranslations}
