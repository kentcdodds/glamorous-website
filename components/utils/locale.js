const {supportedLocales, fallbackLocale} = require('../../config.json')

let {LOCALE: lang} = process.env
if (!supportedLocales.includes(lang)) {
  lang = fallbackLocale
}
module.exports = lang
