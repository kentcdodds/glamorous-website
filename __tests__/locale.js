/* global it, expect, describe */

import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import {matcher, serializer} from 'jest-glamor-react'
import {withLocale} from '../components/locale'

expect.addSnapshotSerializer(serializer)
expect.extend(matcher)

describe('Testing locale', () => {
  it('Renders withLocale', () => {
    const SampleComponent = () =>
      <withLocale><div sample="prop">Hello World</div></withLocale>

    expect(toJson(shallow(<SampleComponent />))).toMatchSnapshotWithGlamor(
      'shallow',
    )
  })
})
