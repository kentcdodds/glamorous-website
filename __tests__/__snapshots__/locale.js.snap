// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow 1`] = `
<withLocale>
  <div
    sample="prop"
  >
    Hello World
  </div>
</withLocale>
`;
